core/templates/expressions/expression-syntax-tree.service.ts

```diff
@@ -390,54 +389,57 @@ export class ExpressionSyntaxTreeService {
       }
     },
     floor: {
-      eval: (args: Array<any>): number => {
+      eval: (args: Array<number|string>): number => {
         this.verifyNumArgs(args, 1);
         var numericArgs = this._coerceAllArgsToNumber(args);
         return Math.floor(numericArgs[0]);
       },
-      getType: (args: Array<any>): string => {
+      getType: (args: Array<number|string>): string => {
         this.verifyNumArgs(args, 1);
         this._verifyArgTypesMatchExpectedType(args,
           AppConstants.PARAMETER_TYPES.REAL);
         return AppConstants.PARAMETER_TYPES.REAL;
       }
     },
     pow: {
-      eval: (args: Array<any>): number => {
+      eval: (args: Array<number|string>): number => {
```

Comment:
Why we need `string` here? On which line exactly does marking argument as `Array<number>` throw error?

Reply:
It used to throw on `verifyNumArgs` method because I've set the args on that method to `Array<number|string>`. Do we actually have a specification for these operators, if, pow, log and abs? Otherwise, I'll find it hard to figure out what `+` should evaluate. 
So far what I thought of these use cases: `3 + 'text as string'` or `-2` or `'hello world!'` which turns out that I needed to set `args` to `Array<number|string>` same for `pow`. 

Correct me if I'm wrong 😲

Reply:
@bansalnitish I've sent you an invitation on hangouts

Reply:
1. Correct me if I'm wrong -- `verifyNumArgs` expects args to be `string` and Math.functions expect args to be just `number`. However it should be just number for Math related functions hence you added a + sign.

2. What does _coerceAllArgsToNumber(args) expect args to be?

3. I didn't get any invitation on Hangouts.

Reply:
@bansalnitish <img width="395" alt="Screenshot 2020-03-25 at 17 03 13" src="https://user-images.githubusercontent.com/11369764/77557875-8acec300-6eba-11ea-8a8c-5043dc014c4b.png">

Reply:
@bansalnitish  
2. `_coerceAllArgsToNumber(args: Array<any>)`
When I set it to `Array<number|string>` or only `Array<string>` I get an error like this because `_coerceToNumber()` originalValue expects to be string.

Reply:
Okay I understand the issue. Let's mark it Array<number|string> as we did initially and use the + sign that you've done. 

I'm unable to find your invite. I checked my account searching your name and even on hangouts.google.com. Am I missing something?

Reply:
@bansalnitish, try clicking this **Invitations**
![image](https://user-images.githubusercontent.com/30312043/77854164-356e1b00-7206-11ea-9325-a5f064221577.png)